src/Draco.Compiler.Tests/Decompilation/CompiledLibrary.cs

```diff
@@ -0,0 +1,130 @@
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Reflection;
+using System.Reflection.PortableExecutable;
+using System.Runtime.Loader;
+using Draco.Compiler.Api;
+using Draco.Compiler.Api.Semantics;
+using Draco.Compiler.Api.Syntax;
+using Draco.Compiler.Internal.Codegen;
+using Draco.Compiler.Internal.Symbols.Metadata;
+using Draco.Compiler.Tests.Utilities;
+
+namespace Draco.Compiler.Tests.Decompilation;
+
+internal sealed class CompiledLibrary
+{
+    public CompiledLibrary(Compilation compilation, MetadataCodegen codeGen, ReadOnlyMemory<byte> assemblyBytes)
+    {
+        Compilation = compilation;
+        Codegen = codeGen;
+        AssemblyBytes = assemblyBytes;
+    }
+
+    public Compilation Compilation { get; }
+    public MetadataCodegen Codegen { get; }
+    public ReadOnlyMemory<byte> AssemblyBytes { get; }
+
+    private PEReader GetPeReader() => _reader ??= new PEReader(new ReadOnlyMemoryStream(AssemblyBytes));
+    private PEReader? _reader;
+
+    public Assembly GetLoadedAssembly() => _loadedAssembly ??= LoadAssembly();
+    private Assembly? _loadedAssembly;
+
+    private Assembly LoadAssembly()
+    {
+        Debug.Assert(_loadedAssembly is null);
+
+        var loadContext = new AssemblyLoadContext("testLoadContext");
+
+        var asm = loadContext.LoadFromStream(new ReadOnlyMemoryStream(AssemblyBytes));
+        return asm;
+    }
+
+    public void AssertIL(string memberName, string il)
+    {
+        var peReader = GetPeReader();
+        var compiledAssemblyReference = MetadataReference.FromPeReader(peReader);
+        var c = Compilation.Create(
+                    ImmutableArray<SyntaxTree>.Empty,
+                    Compilation.MetadataReferences.Add(compiledAssemblyReference));
+
+
+        // TODO: very clunky lookup which won't work if multiple method are defined
+        var member =
+            Compilation
+                .GetSemanticModel(Compilation.SyntaxTrees[0])
+                .GetAllDefinedSymbols(Compilation.SyntaxTrees[0].Root)
+                .OfType<SymbolBase>()
+                .Single(s => s.Symbol is Draco.Compiler.Internal.Symbols.FunctionSymbol);
+
+        var func = Assert.IsAssignableFrom<MetadataMethodSymbol>(member);
+
+        var actualIl = CilFormatter.VisualizeIl(this, func, peReader, compiledAssemblyReference.MetadataReader);
+
+        Assert.True(CilSpaceAgnosticStringComparer.Ordinal.Equals(il, actualIl));
+    }
```

Comment:
This thing has bad debugging experience. It just tells that string didn't match, but doesn't say where